### PR TITLE
Adds `forall` method on Sequences.

### PR DESCRIPTION
**Pull request by gsmaverick**

### CODE REVIEW

**Comment by munificent** [comment]
Very nice!

**Comment by gsmaverick** [comment]
I added the two tests you requested.

On the truthiness-negation I changed that line to `if (!f.call(element)) return false` in which case if you return a non-bool object from the `predicate` you get a runtime error. I'll open a separate PR to tackle moving the negation operator up to the object instead, sound good?

**Comment by munificent** [comment]
These look great! I had one more comment that I *thought* I typed up but now don't see. Here it is:

When I first saw this PR, I intuited that `forall` was like `forEach` in JS, et. al. The "for" implies that to me. How do you feel about calling this `all` instead?

**Comment by gsmaverick** [comment]
Good point, what do you think about using `every`?

**Comment by gsmaverick** [comment]
Actually I take that back, looks like a number of languages already use `all`. Will update.

**Comment by gsmaverick** [comment]
@munificent Renamed should be good to go now.

**Comment by munificent** [comment]
Awesome, thanks!

File: doc/site/core-library.markdown [hunk 1]
```diff
@@ -313,6 +313,10 @@ Removes all items from the list.
 
 The number of items in the list.
 
+### **forall(predicate)**
```
Comment (lluchs) [inline doc/site/core-library.markdown]: You forgot to rename here.

**Pull request merged**